source/common/http/http2/codec_impl.cc

```diff
@@ -573,6 +589,30 @@ ConnectionImpl::~ConnectionImpl() {
   nghttp2_session_del(session_);
 }
 
+void ConnectionImpl::sendKeepalive() {
+  SystemTime now = connection_.dispatcher().timeSource().systemTime();
+  uint64_t ms_since_epoch =
+      std::chrono::duration_cast<std::chrono::milliseconds>(now.time_since_epoch()).count();
+  ENVOY_CONN_LOG(trace, "Sending keepalive PING {}", connection_, ms_since_epoch);
+
+  // The last parameter is an opaque 8-byte buffer, so this cast is safe.
+  int rc = nghttp2_submit_ping(session_, 0 /*flags*/, reinterpret_cast<uint8_t*>(&ms_since_epoch));
+  ASSERT(rc == 0);
```

Comment:
How confident are you that this is always true?

Reply:
I see, this is only NOMEM. In that case, I think a `RELEASE_ASSERT` is warranted.

Reply:
I copied the pattern from elsewhere in the file; it's pervasive. I personally have no confidence that it's true.